server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -210,34 +215,100 @@ public AsyncJobStatus queueJob(JobBuilder builder) {
+        }
+        catch (JobStateManagementException e) {
+            log.error("Unable to update state for job: {}; leaving job in its previous state for " +
+                "state resync upon execution", status.getName());
+
+            // We were unable to update the state from CREATED->QUEUED, but we were able to send
+            // the message to Artemis. We *should* be fine once the job executes and corrects
+            // itself. The job will skip the QUEUED state and transition right into RUNNING, but
+            // that's fine... I guess.
+
+            // Manually update the state just to be certain. This won't persist, but at least our
+            // output will be consistent.
+            status.setState(JobState.QUEUED);
+        }
+        catch (JobMessageDispatchException e) { // Temporary exception branch
+            log.error("Unable to dispatch job message for new job: {}; deleting job and returning",
+                status.getName());
+
+            // We created the job, but were unable to send the job to Artemis. The job is dead
+            // at this point. We *could* retry, but at the time of writing, we have no mechanism
+            // for enabling retry or async messaging (comes with scheduling). As such, we'll
+            // kill the job
+
+            this.jobCurator.delete(status);
 
-            // Update the job state to QUEUED
-            job = job.setState(JobState.QUEUED);
-            this.jobCurator.merge(job);
+            status.setState(JobState.FAILED);
+            status.setJobResult(e);
         }
         catch (Exception e) {
-            log.warn("Error sending async job message.", e);
+            log.error("Unexpected exception occurred while queueing job: {}", status.getName(), e);
 
-            // We couldn't send the message so discard the job status as it is no longer relevant.
-            if (job != null && job.getId() != null) {
-                this.jobCurator.delete(job);
-            }
+            // Uh oh. If we're here, something very very bad has happened. This probably indicates
+            // the database is unavailable or we, otherwise, cannot persist the AsyncJobStatus
+            // instance. We'd delete it, but that'll likely fail, too. Best thing to do here is
+            // just return the job status with the FAILED info.
+
+            // If this occurs do to some other unexpected failure, we'll have some state cleanup
+            // to deal with, probably.
 
-            throw new RuntimeException("Error sending async job message.", e);
+            status.setState(JobState.FAILED);
+            status.setJobResult(e);
         }
 
         // Done!
-        return job;
+        return status;
+    }
+
+    /**
+     * Creates and dispatches a job message for the given job status, then updates the state of
+     * the job to QUEUED.
+     *
+     * TODO: Remove this and integrate it with the scheduling bits so that it can be retried should
```

Comment:
What does 'integrate it with the scheduling bits' mean?

Reply:
In our discussion you had mentioned doing some kind of retry on the message sending to try to avoid instantly failing out of jobs. Rather than worry about rolling our own persistent retry stuff, I figured we could just as easily have another Quartz-backed task here that would try some number of times to send the message.

Reply:
Ah, I see. What if we just go with, if you can't reschedule a retry, the job is done. Basically try again. I can't see this happening very often, so perhaps this approach would be simpler and not add more complexity. Thoughts?

Reply:
We can always add it in later if it becomes a problem. I'm also not sure of how often that retry logic would be hit anyway.